src/status.rs

```diff
@@ -516,17 +518,48 @@ fn compute_status_hash(history: &[HistoryEntry]) -> Option<StatusHash> {
     Some(StatusHash::from_engine(engine))
 }
 
+struct FilteredTx<T> {
+    tx: Transaction,
+    txid: Txid,
+    pos: usize,
+    proof: Option<Proof>,
+    result: Vec<T>,
+}
+
 fn filter_block_txs<T: Send>(
     block: Block,
-    f: impl Fn(usize, Transaction, &[Txid]) -> Option<T> + Sync,
-) -> Vec<T> {
-    let txids: Vec<Txid> = block.txdata.par_iter().map(|tx| tx.txid()).collect();
+    map_fn: impl Fn(&Transaction) -> Vec<T> + Sync,
+    cache_proofs: bool,
+) -> impl Iterator<Item = FilteredTx<T>> {
```

Comment:
Is there some reason this can't return `impl ParallelIterator`?

Reply:
IIUC, we can't use `for_each()` [here](https://github.com/romanz/electrs/pull/610/files#diff-0bfaab7c77d86eb41c2fe32c5e5df509fa02ab04c97b0ce3216911afaae0d67cR349), since it requires sharing mutable references to `block_entries` and `outpoints` between multiple threads.
Please let me know if there is a better way to do it :)

Reply:
May be worth benchmarking and comparing with [lock-free map](https://docs.rs/lockfree/0.5.1/lockfree/map/struct.Map.html)

Or maybe it's possible to somehow collect the generated items with a few more combinators?

Reply:
> May be worth benchmarking and comparing with [lock-free map](https://docs.rs/lockfree/0.5.1/lockfree/map/struct.Map.html).

TIL, will try :)

> Or maybe it's possible to somehow collect the generated items with a few more combinators?

Sorry, couldn't find the right way for doing it here:
https://docs.rs/rayon/1.5.1/rayon/iter/trait.ParallelIterator.html